src/providers/data_provider_be.c

```diff
@@ -615,6 +620,59 @@ errno_t be_process_init(TALLOC_CTX *mem_ctx,
     return ret;
 }
 
+static void watch_update_resolv(const char *filename, void *arg)
+{
+    int ret;
+
+    DEBUG(SSSDBG_TRACE_FUNC, "Reloading %s.\n", filename);
+
+    ret = res_init();
+    if (ret != 0) {
+        DEBUG(SSSDBG_OP_FAILURE, "Failed to reload %s.\n", filename);
+        return;
+    }
```

Comment:
Related to another `SIGUSR2` comment, I think code should try going online here.
If previously this wasn't happening, then this is a bug, imo.

CC: @sumit-bose, @pbrezina

Reply:
Previously update of `resolv.conf` triggered:
```
data_provider_res_init(TALLOC_CTX *mem_ctx,
                       struct sbus_request *sbus_req,
                       struct be_ctx *be_ctx)
{
    resolv_reread_configuration(be_ctx->be_res->resolv);
    check_if_online(be_ctx, 1);

    return monitor_common_res_init(mem_ctx, sbus_req, NULL);
}
```

`resolv_reread_configuration()` and `check_if_online()` are missing now.

Reply:
Correct. Fixed.

Reply:
I agree with a comment https://github.com/SSSD/sssd/pull/6421#discussion_r1042164720 that `check_if_online()` and `res_init()` should be executed in a different order.

Reply:
Done